### PR TITLE
add cafe editing

### PR DESCRIPTION
**Pull request by MaxAstin**

### CODE REVIEW

File: data/src/main/java/com/bunbeauty/data/model/entity/CafeEntity.kt [hunk 1]
```diff
@@ -6,12 +6,13 @@ import androidx.room.PrimaryKey
 @Entity
 data class CafeEntity(
     @PrimaryKey
-    var uuid: String = "",
+    var uuid: String,
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/model/entity/CafeEntity.kt:9]: can we rewrite to val?

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt [hunk 2]
```diff
@@ -13,52 +15,103 @@ import javax.inject.Inject
...
+    private val cafeDao: CafeDao,
+) : CafeRepo {
 
-    var cafeListCache: List<Cafe>? = null
+    private var cafeListCache: List<Cafe>? = null
 
-    override suspend fun getCafeByUuid(uuid: String): Cafe? =
-        cafeDao.getCafeByUuid(uuid)?.let(cafeMapper::map)
+    override suspend fun getCafeByUuid(uuid: String): Cafe? {
+        return withContext(IO) {
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt:24]: extra withContext

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt [hunk 3]
```diff
@@ -13,52 +15,103 @@ import javax.inject.Inject
...
+        }
     }
 
     override suspend fun getCafeList(cityUuid: String): List<Cafe> {
         val cache = cafeListCache
-        return if (cache == null) {
-            val cafeList = getRemoteCafeList(cityUuid)
-            if (cafeList == null) {
-                getLocalCafeList(cityUuid)
+        return withContext(IO) {
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt:41]: extra withContext

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt [hunk 4]
```diff
@@ -13,52 +15,103 @@ import javax.inject.Inject
...
+            }
+
+            cachedCafe ?: cafeDao.getCafeByUuid(uuid)?.let(cafeMapper::toCafe)
+        }
+    }
 
-    override suspend fun getCafeListByCityUuid(cityUuid: String): List<Cafe> = withContext(IO) {
-        cafeDao.getCafeListByCityUuid(cityUuid).map(cafeMapper::map)
+    override suspend fun getCafeListByCityUuid(cityUuid: String): List<Cafe> {
+        return withContext(IO) {
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt:34]: extra withContext

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt [hunk 5]
```diff
@@ -13,52 +15,103 @@ import javax.inject.Inject
...
+        val patchedCafeServer = foodDeliveryApi.patchCafe(
+            cafeUuid = cafeUuid,
+            patchCafe = patchCafe,
+            token = token,
+        ).dataOrNull() ?: return null
+
+        cafeDao.insert(cafeMapper.toCafeEntity(patchedCafeServer))
+
+        val updatedCafe = cafeMapper.toCafe(patchedCafeServer)
+        cafeListCache = cafeListCache?.map { cachedCafe ->
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt:106]: lest extract fun

File: data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                    nonWorkingDayList
+                }
+            } else {
+                cache
+            }
+        }
+    }
+
+    override suspend fun saveNonWorkingDay(token: String, newNonWorkingDay: NewNonWorkingDay): NonWorkingDay? {
+        return withContext(IO) {
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt:43]: extra withContext

File: data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                nonWorkingDayMapCache[nonWorkingDay.cafeUuid] =
+                    nonWorkingDayMapCache[nonWorkingDay.cafeUuid].orEmpty() + nonWorkingDay
+
+                nonWorkingDay
+            }
+        }
+    }
+
+    override suspend fun updateNonWorkingDay(token: String, uuid: String, isVisible: Boolean): NonWorkingDay? {
+        return withContext(IO) {
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt:59]: extra withContext

File: data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+            foodDeliveryApi.patchNonWorkingDay(
+                token = token,
+                uuid = uuid,
+                patchNonWorkingDay = PatchNonWorkingDayServer(isVisible = isVisible),
+            ).dataOrNull()?.let { nonWorkingDayServer ->
+                val nonWorkingDay = nonWorkingDayMapper.toNonWorkingDay(nonWorkingDayServer)
+                nonWorkingDayDao.insert(nonWorkingDayMapper.toNonWorkingDayEntity(nonWorkingDay))
+                nonWorkingDayMapCache[nonWorkingDay.cafeUuid]?.let { nonWorkingDayList ->
+                    nonWorkingDayMapCache[nonWorkingDay.cafeUuid] = nonWorkingDayList.map { cachedNonWorkingDay ->
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt:68]: lets extract map to fun

File: domain/src/main/java/com/bunbeauty/domain/model/nonworkingday/FormattedNonWorkingDay.kt [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package com.bunbeauty.domain.model.nonworkingday
+
+class FormattedNonWorkingDay(
```
Comment (xidealo) [inline domain/src/main/java/com/bunbeauty/domain/model/nonworkingday/FormattedNonWorkingDay.kt:3]: not data class?

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeAction.kt [hunk 10]
```diff
@@ -0,0 +1,20 @@
+package com.bunbeauty.presentation.feature.editcafe
+
+import com.bunbeauty.presentation.viewmodel.base.Action
+import java.time.LocalDate
+import java.time.LocalTime
+
+sealed interface EditCafeAction : Action {
```
Comment (xidealo) [inline presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeAction.kt:7]: can we move it to data state?

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeEvent.kt [hunk 11]
```diff
@@ -0,0 +1,14 @@
+package com.bunbeauty.presentation.feature.editcafe
+
+import com.bunbeauty.presentation.viewmodel.base.Event
+
+sealed interface EditCafeEvent: Event {
```
Comment (xidealo) [inline presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeEvent.kt:5]: can we move it to data state?

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+            is EditCafeAction.DeleteNonWorkingDay -> {
+                requestConfirmDelete(action.uuid)
+            }
+
+            is EditCafeAction.ConfirmDeleteNonWorkingDay -> {
+                deleteNonWorkingDay(action.uuid)
+            }
+
+            EditCafeAction.BackClick -> {
+                event {
```
Comment (xidealo) [inline presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt:84]: lets rename to addEvent

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+        initialNonWorkingDayDate = getInitialNonWorkingDayDate(),
+        yearRange = getNonWorkingDayYearRange(),
+        minNonWorkingDayDate = getMinNonWorkingDayDate(),
+    )
+) {
+
+    override fun handleAction(action: EditCafeAction) {
+        when (action) {
+            is EditCafeAction.Init -> {
+                state { state ->
```
Comment (xidealo) [inline presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt:54]: letst rename to setState

File: presentation/src/main/java/com/bunbeauty/presentation/viewmodel/base/DataState.kt [hunk 14]
```diff
@@ -0,0 +1,3 @@
+package com.bunbeauty.presentation.viewmodel.base
+
+interface DataState
```
Comment (xidealo) [inline presentation/src/main/java/com/bunbeauty/presentation/viewmodel/base/DataState.kt:3]: can we place all in one file like in client pls 
Event
DataState
ViewDataState
ViewState

File: presentation/src/main/java/com/bunbeauty/presentation/viewmodel/main/MainAction.kt [hunk 15]
```diff
@@ -0,0 +1,14 @@
+package com.bunbeauty.presentation.viewmodel.main
+
+import androidx.navigation.NavController
+import com.bunbeauty.presentation.viewmodel.base.Action
+
+sealed interface MainAction : Action {
```
Comment (xidealo) [inline presentation/src/main/java/com/bunbeauty/presentation/viewmodel/main/MainAction.kt:6]: put it in one class

Event , action and dataState

File: data/src/main/java/com/bunbeauty/data/model/entity/CafeEntity.kt (hunk 1 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/model/entity/CafeEntity.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt (hunk 2 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt (hunk 3 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt (hunk 4 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt (hunk 5 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/CafeRepository.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt (hunk 6 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt (hunk 7 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt]: fixed ✅

File: data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt (hunk 8 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/NonWorkingDayRepository.kt]: fixed ✅

File: domain/src/main/java/com/bunbeauty/domain/model/nonworkingday/FormattedNonWorkingDay.kt (hunk 9 above)
Comment (MaxAstin) [reply domain/src/main/java/com/bunbeauty/domain/model/nonworkingday/FormattedNonWorkingDay.kt]: fixed ✅

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeAction.kt (hunk 10 above)
Comment (MaxAstin) [reply presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeAction.kt]: fixed ✅

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeEvent.kt (hunk 11 above)
Comment (MaxAstin) [reply presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeEvent.kt]: fixed ✅

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt (hunk 12 above)
Comment (MaxAstin) [reply presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt]: fixed ✅

File: presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt (hunk 13 above)
Comment (MaxAstin) [reply presentation/src/main/java/com/bunbeauty/presentation/feature/editcafe/EditCafeViewModel.kt]: fixed ✅

File: presentation/src/main/java/com/bunbeauty/presentation/viewmodel/base/DataState.kt (hunk 14 above)
Comment (MaxAstin) [reply presentation/src/main/java/com/bunbeauty/presentation/viewmodel/base/DataState.kt]: fixed ✅

File: presentation/src/main/java/com/bunbeauty/presentation/viewmodel/main/MainAction.kt (hunk 15 above)
Comment (MaxAstin) [reply presentation/src/main/java/com/bunbeauty/presentation/viewmodel/main/MainAction.kt]: fixed ✅

**Review by xidealo — APPROVED** [review]

**Pull request merged**